Improve Population of Tournament Match from Big Match

Rathoz opened the pull request:
## Summary

Copy more data from the Big Match into the Tournament Page match, so we get one complete record

## How did you test this change?

https://liquipedia.net/valorant/Special:LiquipediaDB/VoganRL/Matches

hjpalpha reviewed (approved):
LGTM

JaspervanRiet reviewed (commented):
Am I right in saying this title/description isn't quite accurate? I was confused because I thought the data already is populated from BigMatch, which as the code shows, is true. This is just adding more data, right?

Rathoz commented:
> Am I right in saying this title/description isn't quite accurate? I was confused because I thought the data already is populated from BigMatch, which as the code shows, is true. This is just adding more data, right?

BigMatch page doesn't have all tournament information, such as tickername, tier, header etc. The match on the tournament page does. On the other hand we have the None-BigMatch version of the match, contained inside the bracket or matchlist, which I'll call the Tournament Match (as it's on the tournament page). The tournament match doesn't by default have the match information from the BigMatch.

There was some existing code retrieving the information from the BigMatch into the Tournament Match, but it was just retrieving a few fields (opponents and the none-json map information) and nothing on the match level. This PR expands it to match and all map information. A clearer summary could have been writing by me, I'll update it.

However, I just realised I want to do one more test before merging this. I'll write when I'm done.

Rathoz commented:
> However, I just realised I want to do one more test before merging this. I'll write when I'm done.

Fixed an issue related to that test case (was copying empty records, overwriting eg. tournament level information). It will now only overwrite a Tournament Match record if the Standalone Match has that information.
Also added some comments, we should in general get better at it.

JaspervanRiet commented on components/match2/wikis/valorant/match_group_input_custom.lua:
```diff
@@ -242,12 +242,31 @@ function matchFunctions.mergeWithStandalone(match)
-	local match2games = standaloneMatch.match2games
-	for i = 1, #match2games do
-		match['map' .. i] = match2games[i]
+	-- Update Maps from the Standalone Match
+	for index, game in ipairs(standaloneMatch.match2games) do
+		-- JSON encoded fields has to be parsed first
+		game.participants = Json.parseIfString(game.participants)
+		game.extradata = Json.parseIfString(game.extradata)
+		-- Save the map into current match object
+		match['map' .. index] = game
```
```suggestion
	-- Update Maps from the Standalone Match
	for index, game in ipairs(standaloneMatch.match2games) do
		game.participants = Json.parseIfString(game.participants)
		game.extradata = Json.parseIfString(game.extradata)
		match['map' .. index] = game
```

No need for these comments.

For the top comment you could probably argue it should be made into a function instead `updateMapsFromStandaloneMatch` but let's keep it this way.

JaspervanRiet reviewed (approved):
LGTM
